Expensify/App
# 3770 multi language improvements

### Details


This PR makes these changes:
1. Hide all the locale switching functionality if user doesn't have `internationalization` beta.
2. Move locale selector from preferences page into a separate component `LocalePicker` so it can be reused across the app.
3. Add `LocalePicker` to sign in page's footer.
4. Small footer redesign according to this [figma](https://www.figma.com/file/E8mgPW9ns9EXr8V2LAKE9B/Sign-Up-Sign-In?node-id=705%3A1156) .
5. Selected locale preference persists through logout.

### Fixed Issues

$ #3770

### Tests


1. While logged in, navigate to preferences page and change language to Spanish.
2. Log out.
3. Verify that the language is still Spanish.
4. Verify that selecting English in the new language selector at the footer, correctly changes the app language.

### QA Steps


1. Verify that there is no language selector on preferences page
2. Log out
3. Verify that there is no language selector on sign-in page's footer.

### Tested On


### Screenshots


#### Web



https://user-images.githubusercontent.com/64093836/126486880-82603cd2-fbce-4c1f-96c7-9cbe5e540b7d.mp4



#### Mobile Web


https://user-images.githubusercontent.com/64093836/126486154-eef3da76-b60f-4da4-a89a-91e74659670d.mp4


#### Desktop


https://user-images.githubusercontent.com/64093836/126486169-74d23fdc-4d17-4e17-9756-ffe2d89d4f2a.mov


#### iOS


https://user-images.githubusercontent.com/64093836/126486329-a9f86920-13c9-4255-ad79-6a4e01fdf583.mp4



#### Android


https://user-images.githubusercontent.com/64093836/126486862-d7c71271-df91-4798-b1cf-67194db714c8.mp4
